torch/utils/_pytree.py

```diff
@@ -706,15 +727,17 @@ def tree_map_only(
     __type_or_types: TypeAny,
     func: FnAny[Any],
     tree: PyTree,
+    *rest: PyTree,
 ) -> PyTree:
-    return tree_map(map_only(__type_or_types)(func), tree)
+    return tree_map(map_only(__type_or_types)(func), tree, *rest)
```

Comment:
It doesn't make sense for tree_map only to accept *rest, right? Because each pytree may have leaves with different types